main.tf

```diff
@@ -1,6 +1,9 @@
 locals {
-  server_properties = join("\n", [for k, v in var.server_properties : format("%s = %s", k, v)])
-  enable_logs       = var.s3_logs_bucket != "" || var.cloudwatch_logs_group != "" || var.firehose_logs_delivery_stream != "" ? ["true"] : []
+  server_properties            = join("\n", [for k, v in var.server_properties : format("%s = %s", k, v)])
+  enable_logs                  = var.s3_logs_bucket != "" || var.cloudwatch_logs_group != "" || var.firehose_logs_delivery_stream != "" ? ["true"] : []
+  enable_client_authentication = var.enable_client_authentication
+  client_authentication        = local.enable_client_authentication ? 1 : 0
```

Comment:
I dont understand the logic either, if client authtentication tls only has one parameter (certificate_authority_arns), why not apply it only if variable like client_authentication_tls_certificate_authority_arns   is setted and avoid the use of client_authentication_enabled?

Reply:
The dynamic block is making use of for_each and deliberately set so it;

1. Defaults to `false`
2. Does not require you to do any check if the variable is set as it's done via the locals' definition (centralised if you will)